drivers/source/obc_sci_io.c

```
@@ -89,3 +89,54 @@ void uartAssertFailed(char *file, int line, char *expr){
     sciPrintf("ASSERTION FAILED: %s, file %s, line %d\r\n",
                             expr, file, line);
 }
+
+uint8_t sciReadByte(void) {
+    configASSERT((UART_READ_REG == sciREG) || (UART_READ_REG == scilinREG));
+    SemaphoreHandle_t mutex = (UART_READ_REG == sciREG) ? sciMutex : sciLinMutex;
+
+    uint8_t RecievedByte;
+
+    if (mutex != NULL){
+        if (xSemaphoreTake(mutex, UART_MUTEX_BLOCK_TIME) == pdTRUE){
+            RecievedByte = sciReceiveByte(UART_READ_REG);
+            xSemaphoreGive(mutex);
+            return RecievedByte;
+        }
+    }
```

Something should be returned if this wasn't successful. To be safe, it may be better to pass in a pointer and store the received byte at that address (basically what we do in `sciRead`). Then, you can return the appropriate error code.

Take a look at the `obc_errors.h` header file for the error codes that are currently defined. You can return the appropriate one from this function. If you need to, you can define new error codes in that header as well.